test/integration/integration_test_fixtures.py

```diff
@@ -70,13 +70,19 @@ def setUp(self):
     """
     assert (os.path.exists(self.nighthawk_test_server_path))
     assert (os.path.exists(self.nighthawk_client_path))
-    self.server_port = self.getFreeListenerPortForAddress(self.server_ip)
-    self.admin_port = self.getFreeListenerPortForAddress(self.server_ip)
-    self.parameters["admin_port"] = self.admin_port
-    self.test_server = NighthawkTestServer(self.nighthawk_test_server_path,
-                                           self.nighthawk_test_config_path, self.server_ip,
-                                           self.server_port, self.ip_version, self.parameters)
-    assert (self.test_server.start())
+    attempts = 10
+    while attempts > 0:
+      self.server_port = self.getFreeListenerPortForAddress(self.server_ip)
+      self.admin_port = self.getFreeListenerPortForAddress(self.server_ip)
+      self.parameters["admin_port"] = self.admin_port
+      self.test_server = NighthawkTestServer(self.nighthawk_test_server_path,
+                                             self.nighthawk_test_config_path, self.server_ip,
+                                             self.server_port, self.ip_version, self.parameters)
+      if self.test_server.start():
```

Comment:
This is OK... but it would be much nicer if we had end-to-end port zero bind and assignment. If you can add a TODO and/or point to a tracking issue that would be fine for now.

Reply:
I ended up reverting this, several tries at getting to the pid/ports in the Docker environment failed. The most successful attempt was with `fuser` but that required docker to run privileged. So I just pushed a change to move to port 0 binding and querying the admin interface for the listener port that we are interested in instead.